projectroles/management/commands/cleanappsettings.py

```diff
@@ -70,4 +71,17 @@ def handle(self, *args, **options):
                     )
                 )
                 s.delete()
+
+        db_settings = AppSetting.objects.filter(
+            project__isnull=False, user__isnull=False
+        )
+        for s in db_settings:
+            if s.project.get_role(s.user, inherited_only=True) is None:
```

Comment:
Here we have a problem. The original spec says: "IF user has no role in project (including inherited)". Here, we set `inherited_only=True` so we only delete settings if there are inherited roles in addition to a local one.

Removing the `inherited_only` arg should fix this.

As a very small note, we can just use `not` instead of `is None` for brevity :)